### DISCUSSION TITLE
同时收集多个文件时，某个日志量少的文件的收集会延迟

### DISCUSSION THREAD

**Discussion by wkai886** [alibaba/ilogtail#870] [Help] [answered]
3个配置文件对应三个文件 info.log， warn.log， error.log 日志的收集。
info 量比较大，warn 中等，error 很少偶尔才会有几条 error，发现 info 和 warn 基本正常， error 的收集会延迟。

**Comment by messixukejia** [comment, marked as answer]
可以检查下最后一行是否换行。

方法：替换下面的a.log为需要检查的文件
test "$(tail -c 1 a.log | wc -l)" -eq 0 && echo "no newline at eof: a.log"

可以帮忙改下文档：https://github.com/alibaba/ilogtail/blob/main/docs/cn/data-pipeline/input/file-log.md，可以加一个常见问题的章节。

方便的话，请把使用场景登记下，多谢。https://github.com/alibaba/ilogtail/issues/693

**Comment by cyberzz-dev** [comment]
遇到同样问题
